core/src/main/java/io/undertow/server/protocol/framed/AbstractFramedStreamSinkChannel.java

```diff
@@ -240,12 +240,12 @@ public void run() {
 
     @Override
     public void shutdownWrites() throws IOException {
+        // Queue prior to shutting down writes, since we might send the write buffer
+        queueFinalFrame();
```

Comment:
This is not the same as before. Before, inside the lock, we would skip queueFinalFrame if (anyAreSet(state, STATE_WRITES_SHUTDOWN) || broken). Now, we are always invoking queueFinalFrame no matter what.

Reply:
That's not the problem @fl4via . I analyzed the code and I can guarantee that behaviour is persisted. This is because 
'broken' flag is also checked inside queueFinalFrame() method. If it is set to true, method will not execute. What about anyAreSet(state, STATE_WRITES_SHUTDOWN) check
that also has no effect on queueFinalFrame() method. It is because STATE_WRITES_SHUTDOWN is 'written' only in shutdownWrites() and queueFinalFrame() methods.

Reply:
@ropalka I see what you mean now, I'm approving this PR.